### PR TITLE
feat: support profiling

### PR DESCRIPTION
**Pull request by nrwiersma**
This adds profiling support via Pyroscope.

### CODE REVIEW

File: profile.go [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+	FlagProfileUploadRate = "profiling.upload-rate"
+	FlagProfilingTags     = "profiling.tags"
+	FlagProfilingTypes    = "profiling.types"
+)
+
+// ProfilingFlags are flags that configure profiling.
+var ProfilingFlags = Flags{
+	&cli.StringFlag{
+		Name: FlagProfilingDSN,
+		Usage: "The address an configuration of the Pyroscope server, in the format " +
```
Comment (Ullaakut) [inline profile.go:39]: I don't understand what you mean. Does this work better?

```suggestion
		Usage: "The address to the Pyroscope server, in the format " +
```

File: profile.go [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+			"'http://basic:auth@server:port?token=auth-token&tenantid=tenant-id'.",
+		EnvVars: []string{"PROFILING_DSN"},
```
Comment (Ullaakut) [inline profile.go:41]: Why not use `strcase.ToSNAKE` ?

**Review by Ullaakut — APPROVED** [review]
Just some nitpicks, otherwise LGTM

File: profile.go (hunk 2 above)
Comment (nrwiersma) [reply profile.go:41]: The current version doesn't at the moment, but the next major will.

**Pull request merged**